src/editor/nodes/FloorPlanNode.js

```diff
@@ -68,6 +90,88 @@ export default class FloorPlanNode extends EditorNodeMixin(FloorPlan) {
     this.maxTriangles = 1000;
     this.forceTrimesh = false;
     this.heightfieldMesh = null;
+    this._navMeshMode = NavMeshMode.Automatic;
+    this._navMeshSrc = "";
+    this.navMesh = null;
+    this.trimesh = null;
+    this.heightfieldMesh = null;
+  }
+
+  get navMeshMode() {
+    return this._navMeshMode;
+  }
+
+  set navMeshMode(value) {
+    if (value === NavMeshMode.Custom) {
+      // Force reloading nav mesh since it was removed and this._navMeshSrc didn't change
+      this.load(this._navMeshSrc, undefined, true).catch(console.error);
+    } else if (this.navMesh) {
+      this.remove(this.navMesh);
+      this.navMesh = null;
+    }
+
+    this._navMeshMode = value;
+  }
+
+  get navMeshSrc() {
+    return this._navMeshSrc;
+  }
+
+  set navMeshSrc(value) {
+    this.load(value).catch(console.error);
+  }
+
+  async load(src, onError, force = false) {
+    const nextSrc = src || "";
+
+    if (nextSrc === this._navMeshSrc && nextSrc !== "" && !force) {
+      return;
+    }
+
+    this._navMeshSrc = nextSrc;
+    this.issues = [];
+
+    if (this.navMesh) {
+      this.remove(this.navMesh);
+      this.navMesh = null;
+    }
+
+    try {
+      const { accessibleUrl } = await this.editor.api.resolveMedia(nextSrc);
+
+      const loader = this.editor.gltfCache.getLoader(accessibleUrl);
+
+      const { scene } = await loader.getDependency("gltf");
+
+      const mesh = scene.getObjectByProperty("type", "Mesh");
+
+      if (!mesh) {
+        throw new Error("No mesh available.");
+      }
+
+      const geometry = mesh.geometry.clone(); // Clone in case the user reuses a mesh for the navmesh.
+      mesh.updateMatrixWorld();
+      geometry.applyMatrix(mesh.matrixWorld);
+
+      this.setNavMesh(new Mesh(geometry, new MeshBasicMaterial({ color: 0x0000ff, transparent: true, opacity: 0.2 })));
+
+      if (this.navMesh) {
+        this.navMesh.visible = this.editor.selected.indexOf(this) !== -1;
```

Comment:
Is this needed? Guessing since you are emitting `selectionChanged` at the end of this function the state should already get updated.

Reply:
Yeah, it looks like `onSelect()` isn't called in response to the `selectionChanged` event :man_shrugging: